src/compiler/checker.ts

```diff
@@ -6625,8 +6625,11 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             }
 
             function isHomomorphicMappedTypeWithNonHomomorphicInstantiation(type: MappedType) {
-                return isMappedTypeWithKeyofConstraintDeclaration(type)
-                    && !(getModifiersTypeFromMappedType(type).flags & TypeFlags.TypeParameter);
+                if (!type.target || !isMappedTypeWithKeyofConstraintDeclaration(type)) {
+                    return false;
+                }
+                const index = getIndexType(getModifiersTypeFromMappedType(type));
+                return !(index.flags & TypeFlags.Index && (index as IndexType).type.flags & TypeFlags.TypeParameter);

```

Comment:
So I was thinking something like
```ts
function isMappedTypeHomomorphic(type: MappedType) {
    return !!getHomomorphicTypeVariable(type);
}

function isHomomorphicMappedTypeWithNonHomomorphicInstantiation(type: MappedType) {
    return !type.target ? false : isMappedTypeHomomorphic(type.target) && !isMappedTypeHomomorphic(type);
}
```

Looks easier to reason about a bit, too, by replacing the complex check I proposed before with a direct `getHomomorphicTypeVariable` call on the type and its' target.

Reply:
nice, this is definitely is easier to reason about 👍 pushed out the suggested change